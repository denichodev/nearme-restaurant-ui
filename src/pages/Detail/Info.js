import React from 'react';
import { string } from 'prop-types';

import s from './styles.css';

class Info extends React.PureComponent {
  render() {
    const { cuisine, phoneNumber, openingHours, address } = this.props;
    return (
      <div className={s.detail}>
        <div className={s.heading}>Detail Info</div>
        <div className={s.detailContent}>
          <div className={s.detailItem}>
            <span>Phone Number</span>
            <span>{phoneNumber}</span>
          </div>
          <div className={s.detailItem}>
            <span>Cuisine</span>
            <span>{cuisine}</span>
          </div>
          <div className={s.detailItem}>
            <span>Opening Hours</span>
            <span>{openingHours}</span>
          </div>
          <div className={s.detailItem}>
            <span>Address</span>
            <span>{address}</span>
          </div>
        </div>
      </div>
    );
  }
}

Info.propTypes = {
  address: string.isRequired,
  cuisine: string.isRequired,
  openingHours: string.isRequired,
  phoneNumber: string.isRequired,
};

export default Info;
