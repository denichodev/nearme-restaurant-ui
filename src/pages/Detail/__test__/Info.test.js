import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Info from '../Info';

describe('<Info />', () => {
  it('renders correctly', () => {
    const props = {
      address:
        'Gandaria City, Lantai Upper Ground, Jl. Sultan Iskandar Muda, Gandaria, Jakarta',
      cuisine: 'Asian',
      openingHours: '10.00 - 22.00',
      phoneNumber: '08980780780',
    }
    const wrapper = shallow(<Info {...props} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
