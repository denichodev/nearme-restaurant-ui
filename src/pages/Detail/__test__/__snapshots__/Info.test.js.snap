// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> renders correctly 1`] = `
<div
  className="detail"
>
  <div
    className="heading"
  >
    Detail Info
  </div>
  <div
    className="detailContent"
  >
    <div
      className="detailItem"
    >
      <span>
        Phone Number
      </span>
      <span>
        08980780780
      </span>
    </div>
    <div
      className="detailItem"
    >
      <span>
        Cuisine
      </span>
      <span>
        Asian
      </span>
    </div>
    <div
      className="detailItem"
    >
      <span>
        Opening Hours
      </span>
      <span>
        10.00 - 22.00
      </span>
    </div>
    <div
      className="detailItem"
    >
      <span>
        Address
      </span>
      <span>
        Gandaria City, Lantai Upper Ground, Jl. Sultan Iskandar Muda, Gandaria, Jakarta
      </span>
    </div>
  </div>
</div>
`;
